components/builder-api/habitat/hooks/run

```diff
@@ -12,11 +12,14 @@ echo "_e{${#title},${#text}}:$title|$text|#api"  >/dev/udp/localhost/8125
 # Wait for the .pem file before starting the service, otherwise the service
 # continually panics in a tight loop, blocking the supervisor from writing
 # out the file.
-while ! [ -f "{{pkg.svc_files_path}}/builder-github-app.pem" ];
-do
-    echo "Waiting for builder-github-app.pem"
+# shellcheck disable=SC2050
+if [[ "{{cfg.api.features_enabled}}" =~ "jobsrv" ]]; then
```

Comment:
I think this is the right thing to do, but I am concerned that we may be assuming the file lives on disk in other places in the code that may cause a panic if it's missing.  Our api test suite wouldn't necessarily catch this because it's all driven by the studio and that expects the file to be present.  

I also can't recall the context on the above comment as to why it would enter a panic loop if the file wasn't present, but I _think_ the on-prem-depot runs without the file present, which contradicts my concern.  

The tests we run in CI generate one for us before starting the test suite (https://github.com/habitat-sh/builder/blob/master/.expeditor/templates/studiorc#L15-L17).  It doesn't need to be valid, just present, to start. This also makes me think our test suite may not hit any code paths that actually exercise the github-api-client(https://github.com/habitat-sh/builder/tree/master/components/github-api-client/src), since any pem file will do.

All this makes me wonder if there's some code chances we could make in the api or the github-api-client to better handle the optional nature of that file, but that would be a larger slice of work and not something for this PR.